### DISCUSSION TITLE
In_combat as a toggle or condition per ability?

### DISCUSSION THREAD

**Discussion by Delekii** [Hekili/hekili#1620] [Q&A] [answered]
Is there a way to disable a specific ability from being recommended under the condition of not being in combat? As an example, avoiding a low impact or instant cast skill as a first cast when aggroing a mob, rather than favouring a higher impact or casted ability that you might get for "free" instead of instantly causing the mob to run at you? 

I've tried a couple of things but I don't seem to have a good enough grasp on how simulationcraft works, or maybe not everything comes across? 

I googled a bit and it seems like I should be able to use in_combat as a condition either in the action profile, or I have also tried putting it into action lists > conditions in the flame shock entries. Is there another more obvious way to do this? 

When I put in_combat in as a condition in the action list page and mouseover it, the condition shows up at "nil", so I assume that either this variable doesn't exist as an option in hekili or it is called something else? 

I also tried using active_enemies, but  it seems like that only works as a counter to check for AoE, and defaults to 1 if 2 or more aren't found rather than 0 if none are found..

**Comment by Hekili** [comment, marked as answer]
If the SimC Wiki lists `in_combat`, can you post a link to that?

`time > 0` will get you the same result.  (When combat hasn't started, your time in combat equals 0.  Time greater than 0 equals you are in combat and have been in combat for X seconds.)

Bear in mind, the addon priority has a `precombat` action list for openers and abilities that should be used prior to combat or to initiate combat.  If you know you always want to start combat with Demonbolt, it's probably better to add Demonbolt (with whatever conditions you're concerned about) to the `precombat` action list rather than adding `time > 0` checks to all the instants you don't want to cast.

**Reply by Delekii** [reply]
I can't remember specifically where I saw it, but https://github.com/simulationcraft/simc/wiki/Conditional-expressions is another example of a resource that shows it.

Edit: I found the original source where I found it mentioned: I was looking through this page: https://www.curseforge.com/wow/addons/hekili/pages/advanced-usage - it mentioned a link to a resource "Available Actions
cf. https://code.google.com/p/simulationcraft/wiki/ActionLists#Available_actions" which is google doc rather than a wiki entry, and in that page I happened across an example actions+=/potion,name=draenic_strength,if=!in_combat|buff.bloodlust.react ```

Sorry, a bit round about but that's how I ended up finding it originally :P

That said, I eventually found another resource that mentioned "combat" rather than in_combat, and combat works just fine as a toggle it seems. 

Edit: deleted the last paragraph because I answered it myself! Thanks for the help
